src/BitStringAddresses/bosefs.jl

```diff
@@ -255,17 +255,23 @@ function excitation(b::B, creations, destructions) where {B<:BoseFS}
 end
 
 """
-    new_address, product = hopnextneighbour(add, chosen)
+    new_address, value = hopnextneighbour(add, chosen, boundary_condition)
 
-Compute the new address of a hopping event for the Bose-Hubbard model. Returns the new
-address and the square root of product of occupation numbers of the involved modes.
+Compute the new address of a hopping event for the Hubbard model. Returns the new
+address and the square root of product of occupation numbers of the involved modes
+multiplied by a term consistent with boundary condition. The following values 
+for `boundary_condition` are supported:
+
+* `:periodic`: hopping over the boundary gives does not change the `value`.
+* `:twisted`: hopping over the boundary flips the sign of the `value`.
+* `:hard_wall`: hopping over the boundary gives a `value` of zero.
+* `θ <: Number`: hopping over the boundary gives a `value` multiplied by ``\\exp(ιθ)`` or ``\\exp(−ιθ)`` depending on the direction of hopping.
```

Comment:
```suggestion
* `θ <: Number`: hopping over the boundary gives a `value` multiplied by ``\\exp(iθ)`` or ``\\exp(−iθ)`` depending on the direction of hopping.
```
This is now very good (concise, complete and accurate). My only suggestion is to use the latin letter "i" rather than "\iota" for the imaginary unit to be consistent with the rest of the documentation.